website/docs/d/networkmanager_core_network_policy_document.html.markdown

```diff
@@ -0,0 +1,255 @@
+    edge_locations {
+      location = "us-east-1"
+      asn      = 64512
+    }
+    edge_locations {
+      location = "eu-central-1"
+      asn      = 64513
+    }
+  }
+
+  segments {
+    name                          = "shared"
+    description                   = "Segment for shared services"
+    require_attachment_acceptance = true
+  }
+  segments {
+    name                          = "prod"
+    description                   = "Segment for prod services"
+    require_attachment_acceptance = true
+  }
+
+  segment_actions {
+    action     = "share"
+    mode       = "attachment-route"
+    segment    = "shared"
+    share_with = ["*"]
+  }
+
+  attachment_policies {
+    rule_number     = 100
+    condition_logic = "or"
+
+    conditions {
+      type     = "tag-value"
+      operator = "equals"
+      key      = "segment"
+      value    = "shared"
+    }
+    action {
+      association_method = "constant"
+      segment            = "shared"
+    }
+  }
+  attachment_policies {
+    rule_number     = 200
+    condition_logic = "or"
+
+    conditions {
+      type     = "tag-value"
+      operator = "equals"
+      key      = "segment"
+      value    = "prod"
+    }
+    action {
+      association_method = "constant"
+      segment            = "prod"
+    }
+  }
+}
+```
+
+`data.aws_networkmanager_core_network_policy_document.test.json` will evaluate to:
+
+```json
+{
+  "version": "2021.12",
+  "core-network-configuration": {
+    "asn-ranges": [
+      "64512-64555"
+    ],
+    "vpn-ecmp-support": false,
+    "edge-locations": [
+      {
+        "location": "us-east-1",
+        "asn": 64512
+      },
+      {
+        "location": "eu-central-1",
+        "asn": 64513
+      }
+    ]
+  },
+  "segments": [
+    {
+      "name": "shared",
+      "description": "Segment for shared services",
+      "require-attachment-acceptance": true
+    },
+    {
+      "name": "prod",
+      "description": "Segment for prod services",
+      "require-attachment-acceptance": true
+    }
+  ],
+  "attachment-policies": [
+    {
+      "rule-number": 100,
+      "action": {
+        "association-method": "constant",
+        "segment": "shared"
+      },
+      "conditions": [
+        {
+          "type": "tag-value",
+          "operator": "equals",
+          "key": "segment",
+          "value": "shared"
+        }
+      ],
+      "condition-logic": "or"
+    },
+    {
+      "rule-number": 200,
+      "action": {
+        "association-method": "constant",
+        "segment": "prod"
+      },
+      "conditions": [
+        {
+          "type": "tag-value",
+          "operator": "equals",
+          "key": "segment",
+          "value": "prod"
+        }
+      ],
+      "condition-logic": "or"
+    }
+  ],
+  "segment-actions": [
+    {
+      "action": "share",
+      "mode": "attachment-route",
+      "segment": "shared",
+      "share-with": "*"
+    }
+  ]
+}
+```
+
+## Argument Reference
+
+The following arguments are optional:
```

Comment:
From the Schema code it seems `core_network_configuration` is set to `Required: true,`. Hence, not all of the arguments are Optional?

Reply:
good catch! im changing to: `The following arguments are available:`